### ISSUE TITLE
Accessing the MainActivity.cs class

### ISSUE THREAD

**Issue by DerickKirkman** [radimitrov/CSharpShellApp#392]
How do I access the main activity class

**Comment by DerickKirkman** [comment]
Any way of doing it

**Comment by jlochoap** [comment]
Same question here. I need access to `OnActivityResult`.

**Comment by radimitrov** [comment]
Access to MainActivity is limited to what is and can be statically exposed via the "Ui" class. It includes OnActivityResult

**Comment by jlochoap** [comment]
> Access to MainActivity is limited to what is and can be statically exposed via the "Ui" class. It includes OnActivityResult

I see, thanks. I ended adding the handler in `ConfigureLifecycleEvents` in the `builder`.
I would also like to use `IntermediateActivity` from `Microsoft.Maui.ApplicationModel` to replicate the FilePicker and FileSaver of the Toolkit (it calls a local OnResult) but it won't let me: the compiler says it is _Inaccessible due to its protection level_... :/

**Comment by radimitrov** [comment]
Yes, that is the preferred way since MAUI came out, though I kept the other way for simplicity and backwards compatibility.

I'm not sure you could use a private activity like that, though it isn't available in the app. 

I might just add CommunityToolkit since it is pretty small and quite useful
